### ISSUE TITLE
[Bug] Path Resolution Failed (ts-node-dev)

### ISSUE THREAD

**Issue by delucca** [yarnpkg/berry#2054]
I'm trying to execute my Typescript project with Yarn 2, but it seems that pnp is not being able to resolve some packages.

Here is my `package.json` file:

```
{
  "name": "execution-mode",
  "version": "0.0.1",
  "description": "Web interface for Bud's execution mode",
  "main": "index.js",
  "scripts": {
    "start:dev": "run-s launch:server",
    "start:debug": "run-s launch:server --inspect",
    "launch:server": "ts-node-dev --project tsconfig.server.json src/index.ts",
    "build": "run-s clean compile copy",
    "compile": "tsc",
    "clean": "rimraf -rf dist",
    "copy": "cp README.md LICENSE package.json dist"
  },
  "author": "delucca <delucca@pm.me>",
  "license": "Apache-2.0",
  "private": true,
  "repository": {
    "type": "git",
    "url": "git+https://github.com/budproj/execution-mode.git"
  },
  "bugs": {
    "url": "https://github.com/budproj/execution-mode/issues"
  },
  "engines": {
    "yarn": ">=2.3.3",
    "node": ">=15.0.0"
  },
  "devDependencies": {
    "@tsconfig/recommended": "^1.0.1",
    "@types/eslint": "^7",
    "@types/eslint-config-prettier": "^6",
    "@types/eslint-plugin-prettier": "^3",
    "@types/koa": "^2.11.6",
    "@types/koa__router": "^8.0.3",
    "@types/node": "^14.14.3",
    "@types/prettier": "^2",
    "@types/react": "^16.9.53",
    "@types/rimraf": "^3",
    "@typescript-eslint/parser": "^4.6.0",
    "eslint": "^7.12.0",
    "eslint-config-prettier": "^6.14.0",
    "eslint-plugin-editorconfig": "^2.1.3",
    "eslint-plugin-prettier": "^3.1.4",
    "lint-staged": "^10.5.0",
    "npm-run-all": "^4.1.5",
    "prettier": "^2.1.2",
    "rimraf": "^3.0.2",
    "ts-node-dev": "^1.0.0"
  },
  "dependencies": {
    "@koa/router": "^9.4.0",
    "chalk": "^4.1.0",
    "koa": "^2.13.0",
    "next": "^9.4.4",
    "react": "^16.14.0",
    "typescript": "^4.0.3"
  }
}
```

Here is the output of `yarn start:dev` with `Yarn 1.22.10`:
```
execution-mode@budproj on  feature/initial-setup [✘!?] via ⬢ v15.0.0 took 2s
❯ yarn start:dev
yarn run v1.22.10
$ run-s launch:server
$ ts-node-dev --project tsconfig.server.json src/index.ts
[INFO] 12:08:43 ts-node-dev ver. 1.0.0 (using ts-node ver. 9.0.0, typescript ver. 4.0.5)
event - compiled successfully
➤ Web server running on: http://local.getbud.co:3000
```

Here is the output of `yarn start:dev` with `Yarn 2.3.3`:
```
execution-mode@budproj on  feature/initial-setup [✘!?] via ⬢ v15.0.0 took 16s
❯ yarn start:dev
/home/odelucca/Code/odelucca/execution-mode@budproj/.pnp.js:15815
    throw firstError;
    ^

Error: Qualified path resolution failed - none of the candidates can be found on the disk.

Source path: /home/odelucca/Code/odelucca/execution-mode@budproj/.yarn/$$virtual/ts-node-dev-virtual-eb89e14dc3/0/cache/ts-node-dev-npm-1.0.0-03ff9cc5a6-3e1064a042.zip/node_modules/ts-node-dev/lib\bin.js
Rejected candidate: /home/odelucca/Code/odelucca/execution-mode@budproj/.yarn/$$virtual/ts-node-dev-virtual-eb89e14dc3/0/cache/ts-node-dev-npm-1.0.0-03ff9cc5a6-3e1064a042.zip/node_modules/ts-node-dev/lib\bin.js
Rejected candidate: /home/odelucca/Code/odelucca/execution-mode@budproj/.yarn/$$virtual/ts-node-dev-virtual-eb89e14dc3/0/cache/ts-node-dev-npm-1.0.0-03ff9cc5a6-3e1064a042.zip/node_modules/ts-node-dev/lib\bin.js.js
Rejected candidate: /home/odelucca/Code/odelucca/execution-mode@budproj/.yarn/$$virtual/ts-node-dev-virtual-eb89e14dc3/0/cache/ts-node-dev-npm-1.0.0-03ff9cc5a6-3e1064a042.zip/node_modules/ts-node-dev/lib\bin.js.json
Rejected candidate: /home/odelucca/Code/odelucca/execution-mode@budproj/.yarn/$$virtual/ts-node-dev-virtual-eb89e14dc3/0/cache/ts-node-dev-npm-1.0.0-03ff9cc5a6-3e1064a042.zip/node_modules/ts-node-dev/lib\bin.js.node

    at internalTools_makeError (/home/odelucca/Code/odelucca/execution-mode@budproj/.pnp.js:15559:34)
    at resolveUnqualified (/home/odelucca/Code/odelucca/execution-mode@budproj/.pnp.js:16592:13)
    at resolveRequest (/home/odelucca/Code/odelucca/execution-mode@budproj/.pnp.js:16616:14)
    at Object.resolveRequest (/home/odelucca/Code/odelucca/execution-mode@budproj/.pnp.js:16688:26)
    at Function.external_module_.Module._resolveFilename (/home/odelucca/Code/odelucca/execution-mode@budproj/.pnp.js:15792:34)
    at Function.external_module_.Module._load (/home/odelucca/Code/odelucca/execution-mode@budproj/.pnp.js:15657:48)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:72:12)
    at node:internal/main/run_main_module:17:47
ERROR: "launch:server" exited with 1.
```

**I've already tried to execute `yarn cache clean --all` and then install the packages again, without success.**

**Comment by merceyz** [comment]
Try making the path relative
```diff
- "launch:server": "ts-node-dev --project tsconfig.server.json src/index.ts",
+ "launch:server": "ts-node-dev --project tsconfig.server.json ./src/index.ts",
```

**Comment by delucca** [comment]
@merceyz even with a relative path, the same issue happens

**Comment by merceyz** [comment]
Try removing the use of `run-s`

**Comment by delucca** [comment]
@merceyz you can clone and run the project itself (it is open source): https://github.com/budproj/execution-mode

It is currently using classic Yarn, but you can update to Berry in order to test.

**Comment by merceyz** [comment]
This is an issue in `ts-node-dev`, seems all versions released after `1.0.0-pre.65` use Windows paths in the bin field which breaks on non-Windows machines. We can probably normalize that on our side just to be safe

**Comment by delucca** [comment]
@merceyz thanks! I've [opened an issue](https://github.com/whitecolor/ts-node-dev/issues/226) there too.

Do you see any workaround? (besides downgrading)

**Comment by merceyz** [comment]
Fixed in #2058, you can grab the bundle from here https://github.com/yarnpkg/berry/actions/runs/332342328 or if you want yarn to clone and build itself run
```
yarn set version from sources --branch 2058
```

**Comment by delucca** [comment]
@merceyz amazing! Now it is working :)

waiting for it to be available in a new CLI release.
